datafusion/functions/src/datetime/to_date.rs

```diff
@@ -47,22 +57,39 @@ impl ToDateFunc {
         match args.len() {
             1 => handle::<Date32Type, _, Date32Type>(
                 args,
+                // Although not being speficied, previous implementations have supported parsing date values that are followed
+                // by a time part separated either a whitespace ('2023-01-10 12:34:56.000') or a 'T' ('2023-01-10T12:34:56.000').
```

Comment:
- Was the previous implementation accepting `<date>T<time>` or `<date>T<anything>`? The comment suggests the former, the code now accepts the latter
- if this is desirable behavior, we should document it (and then we can use a different code comment wording -- not referring to previous implementation as the one constraining us for ever). If this isn't desirable behavior, we should fix it. In any case it might be best to address this as a follow-up issue. @comphead WDYT?